plugins/destination/duckdb/client/transform_gotype.go

```diff
@@ -42,6 +43,20 @@ func getTypedNilValue(arr arrow.Array) any {
 		return nilPtrOf[string]()
 	case *arrow.BinaryType, *arrow.LargeBinaryType, *arrow.FixedSizeBinaryType:
 		return nilPtrOf[[]byte]()
+	case *arrow.ListType:
+		arr := arr.(*array.List)
+		v := getTypedNilValue(arr.ListValues())
+		return reflect.New(reflect.SliceOf(reflect.TypeOf(v))).Interface()
+	case *arrow.LargeListType:
+		arr := arr.(*array.LargeList)
+		v := getTypedNilValue(arr.ListValues())
+		return reflect.New(reflect.SliceOf(reflect.TypeOf(v))).Interface()
+	case *arrow.StructType:
+		// Can't create a Go struct dynamically and maps are unsupported: use string
```

Comment:
can we do JSON?

Reply:
As long as it matches what read() (which uses parquet) expects in type and data, we can try.

Reply:
Turns out ValueStr does exactly that for Struct:

```go
func (a *Struct) ValueStr(i int) string {
	if a.IsNull(i) {
		return NullValueStr
	}

	data, err := json.Marshal(a.GetOneForMarshal(i))
	if err != nil {
		panic(err)
	}
	return string(data)
}
```